cmake/HPX_CheckCXXStandard.cmake

```diff
@@ -37,17 +50,10 @@ if(CMAKE_CXX_STANDARD)
     hpx_error(
       "You've set CMAKE_CXX_STANDARD to ${CMAKE_CXX_STANDARD}, which is less than 17 which is the minimum required by HPX"
     )
-  else()
-    if(HPX_USE_CMAKE_CXX_STANDARD)
-      hpx_warn(
-        "You've set CMAKE_CXX_STANDARD manually, which is not recommended. However, HPX_USE_CMAKE_CXX_STANDARD=ON so we're taking it into account."
-      )
-      set(HPX_CXX_STANDARD ${CMAKE_CXX_STANDARD})
-    else()
-      hpx_error(
-        "You've set CMAKE_CXX_STANDARD manually, which is not recommended. If you really want to set CMAKE_CXX_STANDARD, set HPX_USE_CMAKE_CXX_STANDARD=ON."
-      )
-    endif()
+  elseif(NOT HPX_USE_CMAKE_CXX_STANDARD)
+    hpx_error(
+      "You've set CMAKE_CXX_STANDARD manually, which is not recommended. If you really want to set CMAKE_CXX_STANDARD, set HPX_USE_CMAKE_CXX_STANDARD=ON."
+    )
```

Comment:
If we have `HPX_WITH_CXX_STANDARD` as the normal way to set the standard, we should just set `CMAKE_CXX_STANDARD` unconditionally to `HPX_WITH_CXX_STANDARD`.

Reply:
Hmmm. I thought the idea was not to set the CMAKE_CXX_STANDARD directly to avoid conflicts with depending projects. But I might miss part of the picture.

Reply:
You're probably right... I don't remember the full reason for that. In any case:
1. The error here should probably tell users to use `HPX_WITH_CXX_STANDARD` instead.
2. Since we set `cxx_std_XX` on the targets I don't think we need to set `CMAKE_CXX_STANDARD`. I forgot about this one earlier.

Reply:
Maybe also suggest `HPX_WITH_CXX_STANDARD` in the warning messages of the `HPX_WITH_CXX<xx>` options?